geopandas/tests/test_plotting.py

```diff
@@ -1433,6 +1442,81 @@ def test_polygons_values(self):
         ax.cla()
 
 
+@pytest.mark.skipif(not compat.PANDAS_GE_025, reason="requires pandas > 0.24")
+class TestGeoplotAccessor:
+    def setup_method(self):
+        geometries = [Polygon([(0, 0), (1, 0), (1, 1)]), Point(1, 3)]
+        x = [1, 2]
+        y = [10, 20]
+        self.gdf = GeoDataFrame({"geometry": geometries, "x": x, "y": y})
+        self.df = pd.DataFrame({"x": x, "y": y})
+
+    def compare_figures(self, kind, fig_test, fig_ref, kwargs):
+        """Compare Figures."""
+        ax_pandas_1 = self.df.plot(kind=kind, **kwargs)
+        ax_geopandas_1 = self.gdf.plot(kind=kind, **kwargs)
+        fig_test.subplots().plot(
+            ax=ax_pandas_1,
+        )
+        fig_ref.subplots().plot(ax=ax_geopandas_1)
+
+        ax_pandas_2 = getattr(self.df.plot, kind)(**kwargs)
+        ax_geopandas_2 = getattr(self.gdf.plot, kind)(**kwargs)
+        fig_test.subplots().plot(ax=ax_pandas_2)
+        fig_ref.subplots().plot(ax=ax_geopandas_2)
```

Comment:
Is this actually a correct way to test? (I don't have experience with the `check_figures_equal` decorator). 
It seems that this is plotting nothing on fig_ref/fig_test (the `plot()` call gets no data, and the `ax` keyword gets silently ignored?)

As a comparison, if I do this:

```
In [1]: import matplotlib.pyplot as plt

In [2]: fig, ax = plt.subplots()

In [3]: ax.plot([1, 2, 3])
Out[3]: [<matplotlib.lines.Line2D at 0x7f1907631dc0>]

In [4]: fig_ref = plt.figure()

In [5]: fig_ref.subplots().plot(ax=ax)
Out[5]: []

In [6]: plt.show()
```

The second figure `fig_ref` is empty (and doesn't contain the plot from the previously created `ax`)

Reply:
Potentially relevant SO answer about how to move an Axes object from one figure to another: https://stackoverflow.com/a/46906599/653364

Another option (and probably the better/easier one) is to let geopandas directly plot on those figures from the decorator. Something like:

```
ax1 = fig_ref.subplots()
self.df.plot(kind=kind, ax=ax1, **kwargs)
ax2 = fig_test.subplots()
self.gdf.plot(kind=kind, ax=ax2, **kwargs)
```

Reply:
Nice ! I did not know about this, I have updated the tests to use geopandas plots on the created figures. it seems like the cleaner option as well